### PR TITLE
Improve HA config, and json validation

### PR DESCRIPTION
**Pull request by pegtrifork**

### CODE REVIEW

File: charts/flink-job/values.schema.json [hunk 1]
```diff
@@ -22,18 +22,27 @@
             "properties": {
               "flinkConfiguration": {
                 "required": [
-                  "state.savepoints.dir"
+                  "state.savepoints.dir",
+                  "state.checkpoints.dir",
+                  "high-availability.storageDir"
```
Comment (rkthtrifork) [inline charts/flink-job/values.schema.json:27]:
```suggestion
```

File: charts/flink-job/values.schema.json [hunk 2]
```diff
@@ -61,50 +67,44 @@
             "properties": {
               "flinkConfiguration": {
                 "required": [
-                  "state.checkpoints.dir"
+                  "state.checkpoints.dir",
+                  "high-availability.storageDir"
```
Comment (rkthtrifork) [inline charts/flink-job/values.schema.json:71]:
```suggestion
```

**Review by rkthtrifork — CHANGES_REQUESTED** [review]
i dont understand why exclusiveMinimum means we need the storage section

**Comment by pegtrifork** [comment]
> i dont understand why exclusiveMinimum means we need the storage section

**Comment by pegtrifork** [comment]
> i dont understand why exclusiveMinimum means we need the storage section

It the best way I could find to check for a non empty string. We cannot use "required" since the storage section is already present in values.yaml with an empty string to be able to generate helm docs

**Pull request merged**